integration-cli/docker_cli_service_logs_experimental_test.go

```diff
@@ -49,6 +51,119 @@ func (s *DockerSwarmSuite) TestServiceLogs(c *check.C) {
 	}
 }
 
+// countLogLines returns a closure that can be used with waitAndAssert to
+// verify that a minimum number of expected container log messages have been
+// output.
+func countLogLines(d *daemon.Swarm, name string) func(*check.C) (interface{}, check.CommentInterface) {
+	return func(c *check.C) (interface{}, check.CommentInterface) {
+		out, err := d.Cmd("service", "logs", "-t", name)
+		c.Assert(err, checker.IsNil)
+		lines := strings.Split(strings.TrimSpace(out), "\n")
+		return len(lines), check.Commentf("output, %q", string(out))
+	}
+}
+
+func (s *DockerSwarmSuite) TestServiceLogsCompleteness(c *check.C) {
+	testRequires(c, ExperimentalDaemon)
+	d := s.AddDaemon(c, true, true)
+
+	name := "TestServiceLogsCompleteness"
+
+	// make a service that prints 6 lines
+	out, err := d.Cmd("service", "create", "--name", name, "busybox", "sh", "-c", "for line in $(seq 1 6); do echo log test $line; done; sleep 100000")
+	c.Assert(err, checker.IsNil)
+	c.Assert(strings.TrimSpace(out), checker.Not(checker.Equals), "")
+
+	// make sure task has been deployed.
+	waitAndAssert(c, defaultReconciliationTimeout, d.CheckActiveContainerCount, checker.Equals, 1)
+	// and make sure we have all the log lines
+	waitAndAssert(c, defaultReconciliationTimeout, countLogLines(d, name), checker.Equals, 6)
+
+	args := []string{"service", "logs", name}
+	cmd := exec.Command(dockerBinary, d.PrependHostArg(args)...)
+	r, w := io.Pipe()
+	cmd.Stdout = w
+	cmd.Stderr = w
+	c.Assert(cmd.Start(), checker.IsNil)
+
+	reader := bufio.NewReader(r)
+	// TODO(dperny) are logs guaranteed to come back in order?
```

Comment:
Someone, I cannot remember who, has mentioned that they have seen engine-side logs come back in the wrong order. I don't have proof, this is something like a third-hand account, i'm not sure if it's a bug or a consequence of some technical limitation, and i'm not even sure it's real. would it be alright for me to leave these TODOs in, in case the tests fail later, so someone might know why?